metrics/prometheus.go

```diff
@@ -2,21 +2,45 @@ package metrics
 
 import (
 	"flag"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"log"
 	"net/http"
 	"strconv"
 )
 
 type PrometheusConfig struct {
+	//metricPort to which the metrics would be sent
 	metricPort int
+
+	//metric collector
+	offsetMetrics *prometheus.GaugeVec
 }
 
-func (p *PrometheusConfig) init(){
+func (p *PrometheusConfig) init() {
 	//The metrics can be fetched by a Get request from the http://localhost:9999/metrics end point
 	go func(config *PrometheusConfig) {
 		addr := flag.String("listen-address", ":"+strconv.Itoa(config.metricPort), "The address to listen on for HTTP requests.")
 		http.Handle("/metrics", promhttp.Handler())
 		log.Fatal(http.ListenAndServe(*addr, nil))
 	}(p)
```

Comment:
init is a special function that is executed before main and is ideally used to initialise program, in our case we intend to start the server and register our metrics. I recommend we change this name along with signature to suite the interface.
```suggestion
func (p PrometheusRegistry) start(config interface{}) {
        pConfig, _ := config.(PrometheusConfig)
	//The metrics can be fetched by a Get request from the http://localhost:9999/metrics end point
	go func(config PrometheusConfig) {
		addr := flag.String("listen-address", ":"+strconv.Itoa(config.metricPort), "The address to listen on for HTTP requests.")
		http.Handle("/metrics", promhttp.Handler())
		log.Fatal(http.ListenAndServe(*addr, nil))
	}(pConfig)
```